operators/wireguard/internal/controllers/cluster-connection/dns-util.go

```diff
@@ -0,0 +1,75 @@
+package cluster_connection
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Integration test for `getCorednsConfig` function is required.

Given the critical role of the `getCorednsConfig` function in generating the CoreDNS configuration, an integration test that verifies the correct generation of the DNS config based on different cluster connection scenarios would be valuable.